### PR TITLE
Add Tag component

### LINKED ISSUE #65: Tags
- Tags can be clickable & link to something
- They can be used everywhere
- When applied to search and filter, they can be dismissible
- Tags in the search can be predefined and selectable. Results in filtering with predefined tags

- [x] Icon
- [x] Text
- [x] Dismiss action

### PR DESCRIPTION
**Pull request by cristianoliveira**
Changes

Adds new tag component.

Closes #65

### CODE REVIEW

File: src/components/Tag/Tag.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+const CloseButton = styled(DefaultCloseButton)(closeButtonStyles);
+
+const tagStyles = ({ selected, theme }) => css`
+  label: tag;
+  padding: ${theme.spacings.bit} ${theme.spacings.kilo};
+  border-radius: ${theme.borderRadius.mega};
+  border: 1px solid ${theme.colors.n300};
+  background-color: ${selected ? theme.colors.b500 : theme.colors.n100};
+  color: ${selected && theme.colors.white};
```
Comment (connor-baer) [inline src/components/Tag/Tag.js]: We've made it a convention to isolate styles that are related to a specific state (in this case `selected`). So this should be: 

```js
const tagBaseStyles = ({ theme }) => css`
  label: tag;
  padding: ${theme.spacings.bit} ${theme.spacings.kilo};
  border-radius: ${theme.borderRadius.mega};
  border: 1px solid ${theme.colors.n300};
  background-color: ${theme.colors.n100};
  ...
}`;

const tagSelectedStyles = ({ selected, theme }) => selected && css`
  label: tag--selected;
  background-color: ${theme.colors.b500};
  color: ${theme.colors.white};
}`;
```

This has the advantage of making the styles more composable and mostly, enables us to extract the correct class names during static style extraction (in this case `tag--selected`).

File: src/components/Tag/Tag.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  `;
+
+const iconStyles = ({ selected, theme }) => css`
+  label: tag__icon;
+  vertical-align: -2px;
+  margin-right: ${theme.spacings.bit};
+
+  > svg {
+    fill: ${selected && theme.colors.white};
+  }
```
Comment (connor-baer) [inline src/components/Tag/Tag.js]: Same here.

File: src/components/Tag/Tag.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+`;
+
+const IconContainer = styled('span')(iconStyles);
+const TagElement = styled('span')(tagStyles, tagClickableStyles);
+
+const allowIcon = (onRemove, icon) => {
+  if (onRemove && icon) {
+    console.warn("When tag has onRemove prop it doesn't allow icon.");
```
Comment (connor-baer) [inline src/components/Tag/Tag.js]: Hm, this doesn't feel right to me. Ideally, this should go into the PropTypes as a custom prop checker. WDYT?

File: src/components/Tag/Tag.js (hunk 1 above)
Comment (cristianoliveira) [reply src/components/Tag/Tag.js]: Yeah, that makes sense. Let me change that

File: src/components/Tag/Tag.js (hunk 3 above)
Comment (cristianoliveira) [reply src/components/Tag/Tag.js]: Also true! Totally forgot the custom props checker option

File: src/components/Tag/Tag.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+  cursor: default;
+`;
+
+const tagSelectedStyles = ({ selected, theme }) =>
+  selected &&
+  css`
+  label: tag--selected;
+  background-color: ${theme.colors.b500};
+  color: ${theme.colors.white};
+}`;
```
Comment (mlent) [inline src/components/Tag/Tag.js]: Is this a redundant curly brace?

File: src/components/Tag/Tag.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+  /**
+   * Makes the tag selected.
+   */
+  selected: PropTypes.bool
+};
+
+Tag.defaultProps = {
+  children: null,
+  icon: null,
+  onRemove: null,
```
Comment (mlent) [inline src/components/Tag/Tag.js]: Should onRemove be a no-op function rather than null?

Comment (mlent) [reply src/components/Tag/Tag.js]: Ah, or you're checking this for being truthy in order to show the X. Then maybe it shouldn't have a default prop at all? Same for `icon`?

File: src/components/Tag/Tag.js (hunk 4 above)
Comment (cristianoliveira) [reply src/components/Tag/Tag.js]: Yeap. Fixed

**Pull request merged**